src/tools/compiletest/src/runtest.rs

```diff
@@ -379,7 +381,7 @@ impl<'test> TestCx<'test> {
         }
     }
 
-    fn run_rpass_test(&self) {
+    fn run_pass_test(&self, run: bool) {
         let emit_metadata = self.should_emit_metadata(self.pass_mode());
         let should_run = self.run_if_enabled();
         let proc_res = self.compile_test(should_run, emit_metadata);
```

Comment:
This is passing misleading input into compile_test via should_run, I'm not sure if we should try to reuse run_pass_test across run and not-run modes like this. At minimum coupling `run` into `should_run` probably is necessary.